internal/sessiontest/helper_servers_test.go

```diff
@@ -185,6 +185,9 @@ func chainedServerHandler(t *testing.T, jwtPubKey *rsa.PublicKey) http.Handler {
 		})
 		require.NoError(t, err)
 
+		// give polling time to land in between
```

Comment:
I like that there is a comment. I understand the comment with the current bug in my mind. However, I think why we really have this sleep is because it makes the integration test more realistic. It's a REST call. REST calls cost time. Therefore we wait here before we respond. I'd rather read this in the comment here than one specific use-case that triggered us to add the waiting time.